pco/src/float_quant_utils.rs

```diff
@@ -52,12 +57,82 @@ pub(crate) fn split_latents<F: FloatLike>(page_nums: &[F], k: Bitlen) -> Vec<Vec
   vec![primary, secondary]
 }
 
+pub(crate) fn compute_bid<F: FloatLike>(sample: &[F]) -> Option<Bid<F>> {
+  let (k, freq) = estimate_best_k_and_freq(sample);
+  // Nothing fancy, we simply estimate that quantizing by k bits results in saving k bits per
+  // number, whenever possible. This is based on the assumption that FloatQuant
+  // will usually be used on datasets that are exactly quantized.
+  let bits_saved_per_infrequent_primary = freq * (k as f64);
+  let bits_saved_per_num = sampling::est_bits_saved_per_num(
+    sample,
+    |x| x.to_latent_bits() >> k,
+    bits_saved_per_infrequent_primary,
+  );
```

Comment:
Not for this PR but, are there tests that make sure this estimate is close to the actual saved bits of compression?

Reply:
There are not. We could maybe add these for well-behaved cases? In antagonistic or weird cases our estimates might be substantially off.